### ISSUE TITLE
AzCopy doesn't support sovereign cloud environments

### ISSUE THREAD

**Issue by weikhe** [microsoft/AzureStorageExplorer#1544]
**Storage Explorer Version**:  1.9.0
**Build Number**:  20190701.8
**Platform/OS**:  Windows 10
**Architecture**:  x64
**Regression From**:  

## Bug Description ##

Preview feature "Use AzCopy for Improved Blob Upload and Download" doesn't support sovereign cloud environment (Azure China/ Azure Germany/ Azure US Government). If want to upload files in these environments, it will fail, because AzCopy is still using "core.windows.net" domain for target account, which causes key mismatch and upload fails.

The AzCopy command I copied from my local Storage Explorer:
$env:AZCOPY_CRED_TYPE = "Anonymous";
./azcopy.exe copy "\<file path\>" "https://\<account name\>.blob.**core.windows.net**/\<.......\> --follow-symlinks --recursive --from-to=LocalBlob --blob-type=BlockBlob --put-md5;
$env:AZCOPY_CRED_TYPE = "";

## Steps to Reproduce ##

1. Add a sovereign environment account (Azure China/ Azure Germany/ Azure US Government).
2. Enable preview feature "Use AzCopy for Improved Blob Upload and Download"
3. Upload any files to the account's container

## Expected Experience ##

It should be able to detect account's environment, and change domain in AzCopy commandline accordingly.

## Actual Experience ##

It keeps using "core.windows.net" as account's domain in AzCopy no matter account's real environment, which causes key mismatch and upload fails

**Comment by MRayermannMSFT** [comment]
Hi @weikhe how did you add the account? Was it via:

- accessing an account after signing in
- attaching with name and key
- attaching with a connection string

Thanks!

**Comment by weikhe** [comment]
Hi @MRayermannMSFT 
I add it via **attaching with name and key**: "Use a storage account name and key"

**Comment by jinglouMSFT** [comment]
We may not have the end points when using name and key.  Have you tried to sign-in?

**Comment by weikhe** [comment]
![image](https://user-images.githubusercontent.com/20207363/61259338-614a8d00-a7ac-11e9-995f-2be92666014a.png)
We can select endpoints when using name and key (see attached screenshot), and I'm able to view account's contents, download/upload using non-AzCopy way, so I think it's a bug in this AzCopy feature.

This account is not under my subscription, so I'm using name+key to access. But I think even if I use sign-in way, it would still fail due to wrong domain.

**Comment by MRayermannMSFT** [comment]
Ok, thanks for the followup. I'll get back to you. 😄

**Comment by MRayermannMSFT** [comment]
If you sign in, it will work. For your scenario though, I've identified the change we need to make. Can you wait for 1.10 to come out, or would you like me to give you steps to unblock you now (you'll be editing the app's JavaScript).

**Comment by weikhe** [comment]
Great, I can wait for 1.10, thanks!

**Comment by MRayermannMSFT** [comment]
Fixed merged into master for 1.10.0
